src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -684,4645 +674,8879 @@ private static void VerifyFieldSymbol(FieldSymbol field, string expectedDisplayS
+                //     static R F00(R x, int i) { var y = new R(ref i); return F0(x, y); } // 1
+                Diagnostic(ErrorCode.ERR_EscapeCall, "F0(x, y)").WithArguments("Program.F0(R, R)", "y").WithLocation(11, 61),
+                // (11,67): error CS8352: Cannot use local 'y' in this context because it may expose referenced variables outside of their declaration scope
+                //     static R F00(R x, int i) { var y = new R(ref i); return F0(x, y); } // 1
+                Diagnostic(ErrorCode.ERR_EscapeLocal, "y").WithArguments("y").WithLocation(11, 67),
+                // (15,68): error CS8347: Cannot use a result of 'Program.F0(R, R)' in this context because it may expose variables referenced by parameter 'x' outside of their declaration scope
+                //     static R F10(scoped R x, int i) { var y = new R(ref i); return F0(x, y); } // 2
+                Diagnostic(ErrorCode.ERR_EscapeCall, "F0(x, y)").WithArguments("Program.F0(R, R)", "x").WithLocation(15, 68),
+                // (15,71): error CS8352: Cannot use local 'R' in this context because it may expose referenced variables outside of their declaration scope
+                //     static R F10(scoped R x, int i) { var y = new R(ref i); return F0(x, y); } // 2
+                Diagnostic(ErrorCode.ERR_EscapeLocal, "x").WithArguments("R").WithLocation(15, 71),
+                // (16,68): error CS8347: Cannot use a result of 'Program.F1(R, R)' in this context because it may expose variables referenced by parameter 'x' outside of their declaration scope
+                //     static R F11(scoped R x, int i) { var y = new R(ref i); return F1(x, y); } // 3
+                Diagnostic(ErrorCode.ERR_EscapeCall, "F1(x, y)").WithArguments("Program.F1(R, R)", "x").WithLocation(16, 68),
+                // (16,71): error CS8352: Cannot use local 'R' in this context because it may expose referenced variables outside of their declaration scope
+                //     static R F11(scoped R x, int i) { var y = new R(ref i); return F1(x, y); } // 3
+                Diagnostic(ErrorCode.ERR_EscapeLocal, "x").WithArguments("R").WithLocation(16, 71));
+        }
+
+        [Fact]
+        public void MethodArgumentsMustMatch_01()
+        {
+            var source =
+@"ref struct R
+{
+    public void F0(ref R r) => throw null;
+    public void F1(ref scoped R r) => throw null;
+    public void F2(scoped ref R r) => throw null;
+}
+class Program
+{
+    static void F00(ref R x, ref R y) { x.F0(ref y); }
+    static void F01(ref R x, ref R y) { x.F1(ref y); }
+    static void F02(ref R x, ref R y) { x.F2(ref y); }
+
+    static void F10(ref R x, ref scoped R y) { x.F0(ref y); } // 1
+    static void F11(ref R x, ref scoped R y) { x.F1(ref y); }
+    static void F12(ref R x, ref scoped R y) { x.F2(ref y); } // 2
+
+    static void F20(ref R x, scoped ref R y) { x.F0(ref y); }
```

Comment:
I know it is safe and in keeping with the design, but I find it so confusing that a 'scoped ref' can be passed off to a method which is declared to only accept 'ref'.